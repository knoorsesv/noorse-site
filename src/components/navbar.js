import React, { useEffect, useRef, useState } from 'react'
import { Link } from 'gatsby'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faBars, faTimes } from '@fortawesome/free-solid-svg-icons'
import { Logo } from './images'
import { ExternalLink } from './text'
import { siteMap as defaultSiteMap } from '../env/constants'
import { CoverImage } from './cover-image'

const transition = `transition-all duration-200 ease-in`
const menuBarHeight = 'h-64p sm:h-80p'
const coverSectionHeight = 'h-32v sm:h-48v lg:64v'

export const Navbar = ({ showCoverPhoto, siteMap, initiallyShown }) => {
  const [showNavbar, setShowNavBar] = useState(!showCoverPhoto)

  siteMap = siteMap || defaultSiteMap

  return (
    <React.Fragment>
      <NavContainer
        showCoverPhoto={showCoverPhoto}
        setShowNavBar={setShowNavBar}
      >
        <nav
          className={`
        ${transition}
      z-50 w-full flex
      lg:flex-row-reverse lg:justify-between lg:items-center
      ${
        showNavbar
          ? `${menuBarHeight} fixed bg-green`
          : `${coverSectionHeight} bg-transparent lg:pr-0`
      } 
      `}
        >
          <div
            id="menu-container"
            className={`hidden lg:block 
            ${transition} 
            ${!showNavbar && 'lg:self-start'}
            ${!showNavbar && `lg:mb-10 lg:pl-5 lg:h-38p`}
            `}
          >
            <TopMenu showNavbar={showNavbar} siteMap={siteMap} />
          </div>
          <div
            id="logo-container"
            className={`${transition}
          relative flex justify-center
          ${
            showNavbar
              ? 'justify-start w-1/6 h-full top-8p sm:top-32p left-16p'
              : `w-full h-full lg:w-1/2 p-2 sm:p-6 lg:p-8`
          }
         `}
          >
            <Logo className={`${transition}`} />
          </div>
        </nav>
      </NavContainer>
      <SideBarMenu
        fixedToTop={showNavbar}
        siteMap={siteMap}
        initiallyShown={initiallyShown}
      />
    </React.Fragment>
  )
}

const NavContainer = ({ showCoverPhoto, children, setShowNavBar }) => {
  const ref = useRef(null)

  useEffect(() => {
    const handleScroll = () => {
      if (ref.current) {
        setShowNavBar(
          !showCoverPhoto || ref.current.getBoundingClientRect().top < 0
        )
      }
    }
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('scroll', () => handleScroll)
    }
  }, [showCoverPhoto, setShowNavBar])

  return (
    <section
      ref={ref}
      id="nav-container"
      className={`${
        showCoverPhoto ? coverSectionHeight : menuBarHeight
      } w-full static`}
    >
      {showCoverPhoto ? <CoverImage className={coverSectionHeight} /> : <></>}
      {children}
    </section>
  )
}

const TopMenu = ({ showNavbar, siteMap }) => {
  return [
    <ul
      id="menu"
      key="menu"
      style={{ zIndex: '10', position: 'relative' }}
      className={`list-none h-full flex flex-row justify-between items-center p-4 lg:p-6 lg:mb-0`}
    >
      {siteMap.items.map((item) => (
        <MenuItem key={item.name} item={item} showNavbar={showNavbar} />
      ))}
    </ul>,
    <div
      key="menu-background"
      className={`${showNavbar && 'hidden'} ${transition} shadow-lg`}
      style={{
        position: 'relative',
        top: '-35px',
        borderStyle: 'solid',
        borderWidth: '0 0 40px 20px',
        width: '100%',
        zIndex: '0',
        opacity: '70%',
        borderColor: 'transparent transparent green transparent',
      }}
    />,
  ]
}

const MenuToggle = ({ clickBurger, menuShown }) => {
  return (
    <FontAwesomeIcon
      className={'h-6 w-6'}
      icon={menuShown ? faTimes : faBars}
      onClick={clickBurger}
      id="menu-hamburger"
    />
  )
}

const MenuItem = ({ item, showNavbar }) => {
  return (
    <li
      id={'top-menu-item'}
      className={`relative text-right lg:text-center
      my-2 lg:mx-3
      whitespace-nowrap
      font-medium
      ${transition}
      ${item.subItems && 'group'}`}
    >
      {item.link || item.extLink ? (
        <NavLink activeClassName={`border-b-2`} item={item}></NavLink>
      ) : (
        <span className={'text-white'}>{item.name}</span>
      )}
      {item.subItems && <SubMenuItems fixedToTop={showNavbar} item={item} />}
    </li>
  )
}

const SubMenuItems = ({ item }) => {
  return (
    <div
      id="dropdown"
      className={`
      lg:hidden lg:group-hover:absolute lg:group-hover:flex
      flex flex-col lg:items-start
      w-auto lg:p-6 mr-4
      ${transition}
      bg-green bg-opacity-75`}
    >
      {item.subItems.map((subItem) => {
        return <NavLink item={subItem} key={subItem.name} />
      })}
    </div>
  )
}

const SideBarMenu = ({ fixedToTop, siteMap, initiallyShown }) => {
  const [menuShown, setMenuShown] = useState(initiallyShown)
  const toggleMenuShown = () => {
    setMenuShown(!menuShown)
  }

  return (
    <div
      id="sidebar-container"
      className={`${transition}
      lg:hidden block
      z-50
     h-screen fixed right-0 top-0 p-4 sm:pr-8 
     ${menuShown ? 'w-1/2 sm:w-1/3' : 'w-0'}
     ${menuShown && 'bg-green'}
   `}
    >
      <div
        className={`lg:hidden flex flex-row justify-end items-center 
        mt-2 sm:mt-4 
        ${(menuShown || fixedToTop) && 'text-white'} ${transition}`}
      >
        <MenuToggle clickBurger={toggleMenuShown} menuShown={menuShown} />
      </div>
      <ul
        id="sidebar-menu"
        className={`
            list-none
            ${menuShown ? 'block' : 'hidden'}
            flex flex-col justify-between mt-6 sm:mt-8`}
      >
        {siteMap.items.map((item) => (
          <MenuItem key={item.name} item={item} showNavbar={fixedToTop} />
        ))}
      </ul>
    </div>
  )
}

const NavLink = ({
  item,
  className = 'text-white my-2 text-lg',
  activeClassName = 'border-r-2 pr-2',
}) => {
  return item.link ? (
    <Link
      className={className}
      activeClassName={activeClassName}
      to={item.link}
    >
      {item.name}
    </Link>
  ) : (
    <ExternalLink
      url={item.extLink}
      styled={false}
      icon={false}
      className={className}
      textColor={'text-white'}
      key={item.name}
    >
      {item.name}
    </ExternalLink>
  )
}
