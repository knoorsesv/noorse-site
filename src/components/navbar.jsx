import ctl from '@netlify/classnames-template-literals'
import React, { useEffect, useRef, useState } from 'react'
import { ExternalLink } from './text.jsx'

const transition = `transition-all duration-200 ease-in`
const menuBarHeight = 'h-64p'
export const coverSectionHeight = 'h-32v medium:h-48v large:64v'

const NavSection = ({
  pageHasCoverPhoto,
  children,
  setTopMenuBarShown,
  topMenuBarShown,
  CoverImage,
}) => {
  const ref = useRef(null)

  useEffect(() => {
    const handleScroll = () => {
      if (ref.current) {
        setTopMenuBarShown(
          !pageHasCoverPhoto || ref.current.getBoundingClientRect().top < 0
        )
      }
    }
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('scroll', () => handleScroll)
    }
  }, [pageHasCoverPhoto, setTopMenuBarShown])

  const navContainerClasses = ctl(`
  ${pageHasCoverPhoto ? coverSectionHeight : menuBarHeight}
  ${topMenuBarShown ? 'bg-green' : ''}
  w-full static`)
  return (
    <header ref={ref} id="nav-container" className={navContainerClasses}>
      {pageHasCoverPhoto ? (
        <CoverImage className={ctl(`${coverSectionHeight} absolute w-full`)} />
      ) : (
        <></>
      )}
      {children}
    </header>
  )
}

const MenuItemList = ({
  topMenuBarShown,
  siteMap,
  sideBarMenuShown,
  InfoPageLink,
}) => {
  const ulClasses = ctl(`
  fixed top-0
  list-none z-50
  flex flex-col large:flex-row large:justify-end large:items-center
  space-y-3 large:space-y-0
  pt-16 pr-4 large:pr-6
  w-1/2 medium:w-2/5
  ${transition}
  ${
    sideBarMenuShown
      ? 'h-full bg-green opacity-100 right-0'
      : 'opacity-0 -right-1/2 large:right-0'
  }
  ${
    topMenuBarShown
      ? `h-full large:h-80p bg-green large:opacity-100 right-0 large:w-full large:py-10`
      : 'large:right-0 large:w-auto large:h-16 large:top-6 large:pt-0 large:pl-5 large:opacity-90 large:bg-green'
  }
  `)
  const navClasses = ctl(`
  ${transition}
  fixed w-full
  pr-4
  ${topMenuBarShown ? `z-20 bg-green ${menuBarHeight}` : 'z-50 h-0 p-0'}
  `)

  return (
    <nav className={navClasses}>
      <ul id="menu-list" className={ulClasses}>
        {siteMap.items.map((item) => {
          const liClasses = ctl(`text-right large:text-center
    large:mx-2 large:px-3 large:py-3
  ${transition}
  ${item.subItems && 'group'}`)
          return (
            <li id={'menu-item'} key={item.name} className={liClasses}>
              <NavLink item={item} InfoPageLink={InfoPageLink} />
              {item.subItems && (
                <SubMenuItemList
                  fixedToTop={topMenuBarShown}
                  item={item}
                  InfoPageLink={InfoPageLink}
                />
              )}
            </li>
          )
        })}
      </ul>
    </nav>
  )
}

const SubMenuItemList = ({ item, InfoPageLink }) => {
  const ulClasses = ctl(`
  list-none
  large:hidden large:group-hover:absolute large:group-hover:flex
  flex flex-col space-y-3 large:items-start large:-ml-1
  w-auto large:p-6 mr-4 large:bg-green large:opacity-90
  ${transition}`)

  return (
    <ul id="dropdown" className={ulClasses}>
      {item.subItems.map((subItem) => {
        return (
          <li key={subItem.name}>
            <NavLink item={subItem} InfoPageLink={InfoPageLink} />
          </li>
        )
      })}
    </ul>
  )
}

const NavLink = ({ item, InfoPageLink }) => {
  const className = 'text-white text-lg capitalize'
  if (item.link) {
    return <InfoPageLink className={className} item={item} />
  }

  if (item.extLink) {
    return (
      <ExternalLink
        url={item.extLink}
        styled={false}
        icon={false}
        className={className}
        textColor={'text-white'}
        key={item.name}
      >
        {item.name}
      </ExternalLink>
    )
  }
  return <span className={className}>{item.name}</span>
}

const MenuLogo = ({ topMenuBarShown, Logo }) => {
  const logContainerClasses = ctl(`${transition}
       flex fixed z-20
      ${
        topMenuBarShown
          ? `justify-start w-1/6 top-8 left-16p ${menuBarHeight}`
          : `justify-center w-full top-0 large:w-1/3 p-2 large:p-8 ${coverSectionHeight}`
      }     `)
  return (
    <div id="logo-container" className={logContainerClasses}>
      <Logo className={`${transition}`} />
    </div>
  )
}

const MenuToggle = ({ clickBurger, sideBarMenuShown, topMenuBarShown }) => {
  const toggleWrapperClasses = ctl(`fixed right-0
    large:hidden
    mt-4 mr-3 medium:mt-6 medium:mr-4
    p-2 z-50 text-white
    ${sideBarMenuShown || topMenuBarShown ? '' : 'bg-green'} ${transition}`)
  return (
    <div className={toggleWrapperClasses}>
      <button
        className="bg-transparent text-white"
        onClick={clickBurger}
        aria-label={sideBarMenuShown ? 'Sluit menu' : 'Open menu'}
        id="menu-hamburger"
      >
        {sideBarMenuShown ? <Close /> : <Menu />}
      </button>
    </div>
  )
}

const Close = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    className="h-6 w-6"
    fill="none"
    viewBox="0 0 24 24"
    stroke="currentColor"
    strokeWidth="2"
  >
    <path
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M6 18L18 6M6 6l12 12"
    />
  </svg>
)

const Menu = () => (
  <svg
    xmlns="http://www.w3.org/2000/svg"
    className="h-6 w-6"
    fill="none"
    viewBox="0 0 24 24"
    stroke="currentColor"
    strokeWidth="2"
  >
    <path
      strokeLinecap="round"
      strokeLinejoin="round"
      d="M4 6h16M4 12h16M4 18h16"
    />
  </svg>
)

// todo: just make 2 separate components instead of having a boolean for the cover photo
export const Navbar = ({
  pageHasCoverPhoto = false,
  siteMap,
  InfoPageLink,
  CoverImage,
  Logo,
}) => {
  const [topMenuBarShown, setTopMenuBarShown] = useState(!pageHasCoverPhoto)
  const [sideBarMenuShown, setMenuShown] = useState(false)

  const toggleMenuShown = () => {
    setMenuShown(!sideBarMenuShown)
  }

  return (
    <NavSection
      topMenuBarShown={topMenuBarShown}
      pageHasCoverPhoto={pageHasCoverPhoto}
      setTopMenuBarShown={setTopMenuBarShown}
      CoverImage={CoverImage}
    >
      <MenuItemList
        topMenuBarShown={topMenuBarShown}
        sideBarMenuShown={sideBarMenuShown}
        siteMap={siteMap}
        InfoPageLink={InfoPageLink}
      />
      <MenuLogo topMenuBarShown={topMenuBarShown} Logo={Logo} />
      <MenuToggle
        clickBurger={toggleMenuShown}
        sideBarMenuShown={sideBarMenuShown}
        topMenuBarShown={topMenuBarShown}
      />
    </NavSection>
  )
}
