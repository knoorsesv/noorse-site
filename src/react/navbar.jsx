import ctl from '@netlify/classnames-template-literals'
import { useEffect, useRef, useState } from 'react'
import { ImageWrapper } from '../wrappers/image-wrapper'
import { LinkWrapper } from '../wrappers/link-wrapper.jsx'
import { ChevronDown, ChevronRight, Close, Menu } from './icons/icons.jsx'
import { ExternalLink } from './links/external-link.jsx'
import { Logo } from './logo.jsx'
// import aerial from '../images/noorse_aerial.png?w=600;800;1200&h=200;400;800&format=webp&q=50,100'
import aerial from '../images/noorse_luchtfoto.jpeg?w=600;800;1200&h=400;530;800&format=webp&q=50,100'

const transition = `transition-all duration-200 ease-in`
const menuBarHeight = 'h-64p'
export const coverSectionHeight = 'h-32v medium:h-48v large:64v'

//  topMenuBarShown is a bad name, it means the fixed non transparent navbar should be shown
const NavSection = ({
  pageHasCoverPhoto,
  children,
  setTopMenuBarShown,
  topMenuBarShown,
}) => {
  const ref = useRef(null)

  useEffect(() => {
    const handleScroll = () => {
      if (ref.current) {
        setTopMenuBarShown(
          !pageHasCoverPhoto || ref.current.getBoundingClientRect().top < 0
        )
      }
    }
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('scroll', () => handleScroll)
    }
  }, [pageHasCoverPhoto, setTopMenuBarShown])

  const navContainerClasses = ctl(`
  ${pageHasCoverPhoto ? coverSectionHeight : menuBarHeight}
  ${topMenuBarShown ? 'bg-green' : ''}
  w-full static`)
  return (
    <header ref={ref} id="nav-container" className={navContainerClasses}>
      {pageHasCoverPhoto ? (
        <div className={ctl(`${coverSectionHeight} absolute w-full`)}>
          <ImageWrapper
            id={'background-image'}
            alt={'Luchtfoto Noorse velden'}
            loading={'eager'}
            className="h-full w-full object-cover object-center"
            src={aerial}
          />
        </div>
      ) : (
        <></>
      )}
      {children}
    </header>
  )
}

const MenuItemList = ({ topMenuBarShown, siteMap, sideBarMenuShown }) => {
  const ulClasses = ctl(`
  fixed top-0
  list-none z-50
  flex flex-col large:flex-row large:justify-end large:items-center
  space-y-3 large:space-y-0
  pr-4 large:pr-6
  w-1/2 medium:w-2/5
  ${transition}
  ${
    sideBarMenuShown
      ? 'pt-20 h-full bg-green opacity-100 right-0'
      : 'pt-16 opacity-0 -right-1/2 large:right-0'
  }
  ${
    topMenuBarShown
      ? `h-full large:h-80p bg-green large:opacity-100 right-0 large:w-full large:py-10`
      : 'large:right-0 large:w-auto large:h-16 large:top-6 large:pt-0 large:pl-5 large:opacity-90 large:bg-green'
  }
  `)
  const navClasses = ctl(`
  ${transition}
  fixed w-full
  pr-4
  ${topMenuBarShown ? `z-20 bg-green ${menuBarHeight}` : 'z-50 h-0 p-0'}
  `)

  const InfoPageLink = ({ item, className }) => {
    return (
      <LinkWrapper
        className={className}
        // todo: reenable this when we've moved off gatsby
        // activeClassName={'border-b-2 border-white'}
        href={item.link}
      >
        {item.name}
      </LinkWrapper>
    )
  }

  return (
    <nav className={navClasses}>
      <ul id="menu-list" className={ulClasses}>
        {siteMap.items.map((item) => {
          const [opened, setOpened] = useState(false)
          const liClasses = ctl(`text-right large:text-center
    large:mx-2 large:px-3 large:py-3
  ${transition}
  ${item.subItems && 'group'}`)
          return (
            <li key={item.name} className={liClasses}>
              <NavLink
                item={item}
                InfoPageLink={InfoPageLink}
                onClick={() => setOpened(!opened)}
              />
              {/* todo: icon should be inside button so all is clickable */}
              <span className="inline-flex items-center">
                {sideBarMenuShown ? (
                  opened ? (
                    <ChevronDown
                      className={`ml-1 text-white ${
                        !item.subItems && 'large:hidden'
                      } `}
                    />
                  ) : (
                    <ChevronRight
                      className={`ml-1 text-white ${
                        !item.subItems && 'large:hidden'
                      } `}
                    />
                  )
                ) : (
                  <>
                    <ChevronDown
                      className={`ml-1 text-white ${
                        !item.subItems ? 'hidden' : 'hidden group-hover:inline'
                      } `}
                    />
                    <ChevronRight
                      className={`ml-1 text-white ${
                        !item.subItems ? 'hidden' : 'inline group-hover:hidden'
                      } `}
                    />
                  </>
                )}
              </span>
              {item.subItems && (!sideBarMenuShown || opened) && (
                <SubMenuItemList
                  fixedToTop={topMenuBarShown}
                  item={item}
                  InfoPageLink={InfoPageLink}
                />
              )}
            </li>
          )
        })}
      </ul>
    </nav>
  )
}

const SubMenuItemList = ({ item, InfoPageLink }) => {
  const ulClasses = ctl(`
  list-none
  large:hidden large:group-hover:absolute large:group-hover:flex
  flex flex-col space-y-3 large:items-start large:-ml-1
  w-auto large:p-6 mr-4 large:bg-green large:opacity-90
  ${transition}`)

  return (
    <ul id="dropdown" className={ulClasses}>
      {item.subItems.map((subItem) => {
        return (
          <li key={subItem.name}>
            <NavLink item={subItem} InfoPageLink={InfoPageLink} />
          </li>
        )
      })}
    </ul>
  )
}

const NavLink = ({ item, InfoPageLink, onClick }) => {
  const className = 'text-white text-lg capitalize'
  if (item.link) {
    return <InfoPageLink className={className} item={item} />
  }

  if (item.extLink) {
    return (
      <ExternalLink
        href={item.extLink}
        styled={false}
        icon={false}
        className={className}
        textColor={'text-white'}
        key={item.name}
      >
        {item.name}
      </ExternalLink>
    )
  }
  return (
    <button className={className} onClick={onClick}>
      {item.name}
    </button>
  )
}

const MenuLogo = ({ topMenuBarShown }) => {
  const logContainerClasses = ctl(`${transition}
       flex items-center z-20 
      ${
        topMenuBarShown
          ? `justify-start fixed top-8 left-16p large:left-96p`
          : `justify-center w-full h-full top-0 large:w-1/3 p-2 large:p-8 `
      }     `)
  return (
    <div id="logo-container" className={logContainerClasses}>
      {topMenuBarShown ? (
        <a href="/">
          <Logo
            className={ctl(`${transition} z-30 aspect-square h-auto max-h-full w-full
  ${topMenuBarShown ? `max-w-[64px] ` : `max-w-[200px] medium:max-w-[312px]`}
  `)}
          />
        </a>
      ) : (
        <Logo
          className={ctl(`${transition} z-30 aspect-square h-auto max-h-full w-full
        ${
          topMenuBarShown
            ? `max-w-[64px] `
            : `max-w-[200px] medium:max-w-[312px]`
        }
        `)}
        />
      )}
    </div>
  )
}

const MenuToggle = ({ clickBurger, sideBarMenuShown, topMenuBarShown }) => {
  const toggleWrapperClasses = ctl(`fixed right-0 top-0
    large:hidden
    mt-4 mr-3 medium:mt-6 medium:mr-4
    p-2 z-50 text-white
    ${sideBarMenuShown || topMenuBarShown ? '' : 'bg-green'} ${transition}`)
  return (
    <div className={toggleWrapperClasses}>
      <button
        className="text-white"
        onClick={clickBurger}
        aria-label={sideBarMenuShown ? 'Sluit menu' : 'Open menu'}
        id="menu-hamburger"
      >
        {sideBarMenuShown ? <Close /> : <Menu />}
      </button>
    </div>
  )
}

// todo: just make 2 separate components instead of having a boolean for the cover photo
export const Navbar = ({ pageHasCoverPhoto = false, siteMap }) => {
  const [topMenuBarShown, setTopMenuBarShown] = useState(!pageHasCoverPhoto)
  const [sideBarMenuShown, setMenuShown] = useState(false)

  const toggleMenuShown = () => {
    setMenuShown(!sideBarMenuShown)
  }

  return (
    <NavSection
      topMenuBarShown={topMenuBarShown}
      pageHasCoverPhoto={pageHasCoverPhoto}
      setTopMenuBarShown={setTopMenuBarShown}
    >
      <MenuItemList
        topMenuBarShown={topMenuBarShown}
        sideBarMenuShown={sideBarMenuShown}
        siteMap={siteMap}
      />
      <MenuLogo topMenuBarShown={topMenuBarShown} />
      <MenuToggle
        clickBurger={toggleMenuShown}
        sideBarMenuShown={sideBarMenuShown}
        topMenuBarShown={topMenuBarShown}
      />
    </NavSection>
  )
}
