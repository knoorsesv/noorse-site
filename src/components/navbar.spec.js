import React from 'react'
import renderer from 'react-test-renderer'

import { Navbar } from './navbar'
import { useStaticQuery } from 'gatsby'

useStaticQuery.mockReturnValue({ cover: 'coverImageProps' })

const siteMap = {
  items: [
    { name: 'Top Level Link', link: '/' },
    { name: 'Top Level External Link', extLink: 'http://top-link' },
    {
      name: 'Top Level met Sub Items',
      subItems: [
        { name: 'Sub Item Link', link: '/sub' },
        { name: 'Sub Item Ext Link', extLink: 'http://sub-link' },
      ],
    },
  ],
}

jest.mock('./images', () => ({ Logo: () => <logo-mock /> }))

describe('Navbar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Navbar siteMap={siteMap} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
