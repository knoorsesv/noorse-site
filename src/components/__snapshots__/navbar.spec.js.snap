// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
Array [
  <section
    className="w-full static h-64p sm:h-80p"
    id="header"
  >
    <nav
      className="
        transition-all duration-200 ease-in
      z-50 p-3 w-full flex
      h-64p sm:h-80p 
      fixed bg-green lg:flex-row-reverse lg:justify-between lg:items-center 
      "
    >
      <div
        className="hidden md:block h-64p sm:h-80p transition-all duration-200 ease-in 
            false xl:w-60 
            
            "
        id="menu-container"
      >
        <div
          className="h-full flex flex-row justify-between items-center p-4 lg:p-6"
          id="menu"
          style={
            Object {
              "position": "relative",
              "zIndex": "10",
            }
          }
        >
          <span
            className="mx-3 transition-all duration-200 ease-in"
          >
            <div
              className="relative 
      whitespace-no-wrap
      font-medium
      transition-all duration-200 ease-in
      "
              id="top-menu-item"
            >
              <a
                className="text-white transition-all duration-200 ease-in"
                href="/"
              >
                Top Level Link
              </a>
            </div>
          </span>
          <span
            className="mx-3 transition-all duration-200 ease-in"
          >
            <div
              className="relative 
      whitespace-no-wrap
      font-medium
      transition-all duration-200 ease-in
      "
              id="top-menu-item"
            >
              <span
                className="text-white transition-all duration-200 ease-in"
              >
                Top Level External Link
              </span>
            </div>
          </span>
          <span
            className="mx-3 transition-all duration-200 ease-in"
          >
            <div
              className="relative 
      whitespace-no-wrap
      font-medium
      transition-all duration-200 ease-in
      group"
              id="top-menu-item"
            >
              <span
                className="text-white transition-all duration-200 ease-in"
              >
                Top Level met Sub Items
              </span>
              <div
                className="
      hidden group-hover:absolute group-hover:flex
      flex-col items-start
      w-auto p-6 mr-2
      transition-all duration-200 ease-in
      bg-green bg-opacity-75 "
                id="dropdown"
              >
                <a
                  className="text-white transition-all duration-200 ease-in my-1"
                  href="/sub"
                >
                  Sub Item Link
                </a>
                <a
                  aria-label=""
                  className="false text-white"
                  href="http://sub-link"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Sub Item Ext Link
                </a>
              </div>
            </div>
          </span>
        </div>
        <div
          className="hidden transition-all duration-200 ease-in shadow-lg"
          style={
            Object {
              "borderColor": "transparent transparent green transparent",
              "borderStyle": "solid",
              "borderWidth": "0 0 40px 20px",
              "left": "-25px",
              "opacity": "70%",
              "position": "relative",
              "top": "-35px",
              "width": "130%",
              "zIndex": "0",
            }
          }
        />
      </div>
      <div
        className="transition-all duration-200 ease-in
          relative flex justify-center 
          justify-start w-1/6 h-150 top-8p sm:top-4p md:top-32p left-16p
         "
        id="logo-container"
      >
        <logo-mock />
      </div>
    </nav>
  </section>,
  <div
    className="transition-all duration-200 ease-in
      lg:hidden block
      z-50
     h-screen fixed right-0 top-0 p-4 sm:pr-8 
     w-0
     false
   "
    id="sidebar-menu"
  >
    <div
      className="md:hidden flex flex-row justify-end items-center 
        mt-2 sm:mt-4 
        text-white transition-all duration-200 ease-in"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-bars fa-w-14 h-6 w-6"
        data-icon="bars"
        data-prefix="fas"
        focusable="false"
        id="menu-hamburger"
        onClick={[Function]}
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="
            hidden
            flex flex-col justify-between mt-6 sm:mt-8"
    >
      <span
        className="relative text-right my-1 undefined"
      >
        <a
          className="text-white"
          href="/"
        >
          Top Level Link
        </a>
      </span>
      <span
        className="relative text-right my-1 undefined"
      >
        <span
          className="text-white"
        >
           
          Top Level External Link
           
        </span>
      </span>
      <span
        className="relative text-right my-1 group"
      >
        <span
          className="text-white"
        >
           
          Top Level met Sub Items
           
        </span>
        <div
          className="flex flex-col mr-3 mt-2 space-y-1"
        >
          <a
            className="text-white"
            href="/sub"
          >
            Sub Item Link
          </a>
          <a
            aria-label=""
            className="false text-white"
            href="http://sub-link"
            rel="noopener noreferrer"
            target="_blank"
          >
            Sub Item Ext Link
          </a>
        </div>
      </span>
    </div>
  </div>,
]
`;
